Simplify quick start by making development the default

sampaccoud opened the pull request:
# Purpose

Let's try to make the quickstart as simple as possible

# Proposal

We can simplify 2 things:

- [x] Move all documentation about MiniShift to the same place
- [x] Make VirtualBox the default choice for MiniShift
- [x] Make the development environment default and don't mention it anymore in the quickstart

sampaccoud committed:
✨(env) allow setting environment when running sugar scripts

The env.d/development environment file becomes the default so
this is transparent for developers, especially when they get
started. The scripts have then been refactored so that we
can specify a different environment file by passing a "env-file"
option when running a sugar script.

Fix #17

sampaccoud committed:
♻️(minishift) improve Minishift installation and documentation

Set 4GB minimum RAM when starting MinShift and add explanation
to documentation. We moved all documentation about MiniShift
to the same place in documentation and added some explanations.

sampaccoud committed:
📝(minishift) make VirtualBox the default choice over KVM

Virtualbox is more likely to be installed on developers'
laptops and we encountered networking issues with KVM.

Fix #16

sampaccoud committed:
🚸(minishift) use "cd/cd -" instead of "pushd/popd"

The use case is simple enough to prefer "cd -" as pushd/
popd is not as well known.

jmaupetit commented on bin/_config.sh:
```diff
@@ -2,49 +2,21 @@
-
 # _docker_run: wrap docker run command
 #
 # usage: _docker_run [options] [ARGS...]
```
We need to add the `--env-file` option to the usage.

jmaupetit commented on bin/dev:
```diff
@@ -43,9 +43,9 @@ function _create-kvm-default-network() {
 function _run-minishift() {
 
-    local opts=""
+    local opts="--memory=4GB"
```
:1st_place_medal:

jmaupetit commented on docs/developer_guide/getting_started.md:
```diff
@@ -61,6 +61,36 @@ $ bin/deploy
+You can then specify the environment file when running a sugar script by setting the
+`env-file` option as in the following example:
+
+$ bin/init --env-file=env.d/production
```
It needs to be in a code block.

jmaupetit reviewed (approved):
The getting started documentation needs a cleanup, else :+1: !

sampaccoud committed:
🚚(templates) add missing yml extension to route template

This was generating a bug when deploying the hello
application which was pointing to the file name with the
yml extension.

jmaupetit reviewed (approved):
:+1:
